library/src/main/java/com/google/maps/android/data/Renderer.java

```diff
@@ -351,12 +388,77 @@ public Feature getContainerFeature(Object mapObject) {
     }
 
     /**
-     * Gets the images cache which is needed to download GroundOverlays and Marker Icon images
+     * Gets a cached image at the specified scale which is
+     * needed for Marker icon images
      *
-     * @return mImagesCache
+     * @param url URL to get cached image for
+     * @param scale scale to get image at
+     * @return scaled BitmapDescriptor
      */
-    public LruCache<String, Bitmap> getImagesCache() {
```

Comment:
Yes, I don't believe anyone would actually be using the `Renderer` API directly. Many of these methods should be protected, as they're used only by the `Renderer` subclasses, `KmlRenderer` and `GeoJsonRenderer`. Many of the public methods can't be package-private or protected only because they're used both in `Layer` (same package) and one of the subclasses, which are in different packages.

Code that utilizes the KML and GeoJSON functionality of the library will mostly be interacting with the `Layer` subclasses, `KmlLayer` and `GeoJsonLayer`, which don't even expose their `Renderer` objects directly in order to access those public methods. `Renderer` could almost be an abstract class if the code was refactored some. There's quite a bit of code that's specific to KML and GeoJSON that could also be moved to one or the other subclasses.

I went ahead and reduced the access scope on all the `Renderer` methods that could be, to either package-private or protected. In this particular method's case, a `Renderer` subclass, which I don't expect to exist outside of `KmlRenderer` and `GeoJsonRenderer`, would follow the pattern in these subclasses and call `getCachedMarkerImage()` or `getCachedGroundOverlayImage()` instead of accessing a cache directly as was previously done through this `getImagesCache()` method.

Reply:
Agreed that scoping should be reduced in that class. Thanks for addressing that!